sequencer/finalizer.go

```diff
@@ -81,52 +108,105 @@ func (f *finalizer) Start() {
 		for {
 			tx := f.worker.GetBestFittingTx(wipBatch.remainingResources)
 			if tx != nil {
+				lenOfTxs := len(wipBatch.txs)
+				isFirstTx := lenOfTxs == 1
```

Comment:
```suggestion
				isFirstTx := lenOfTxs == 0
```

At this point, tx is not added yet into `wipBatch.txs`. If `wipBatch.txs == 1` it means that in previous iterations there has been a tx added into the batch, and that tx was actually the first one.

Beyond that we may consider to execute a batch with no txs at all to produce the GER update, and here use always empty hash. We can talk about this later. You can do this in a later PR as here the priority is to define interfaces ASAP

Reply:
Ops, yep mistake here..  from hurrying :D